### DISCUSSION TITLE
Connecting two crazyflies in one platform (Battery compensation)

### DISCUSSION THREAD

**Discussion by FrancisGoncalves** [bitcraze/discussions#1136] [Q&A]
Hello, 

I am working on a platform with two crazyflies connected by a rigid bar. I am using one of the crazyflies to compute all the control signals for the eight motors and send four through UART to the second crazyflie (four 16-bit signals + housekeeping). I know the communication is working fine because I tested it and I was able to achieve stable flight. However, I have noticed that stable flight occurs only when the battery on the main crazyflie is running low, and as soon as I replace the battery, the system is no longer able to take off. I am guessing this has to do with the fact that I am compensating all motor control signals based on one of the batteries. Here are a couple of things I tried and did not work:

(i) Send the uncompensated control signals (four 32-bit signals + housekeeping) and compensate the signals based on the battery of the second crazyflie;
(ii) Bypass the battery compensation function.

I was wondering if you have any ideas on what might be happening and how to solve it.

Thank you!
Francisco

**Comment by tobbeanton** [comment]
Tricky! What do you think about the influence of latency? Could you make an experiment and power both of the crazyflies from the same battery?

**Comment by FrancisGoncalves** [comment]
Since the system already flew stably, I ruled out latency as a problem but maybe something is going on with that. Can one battery handle 8 motors? I thought about that but then I checked the maximum discharge rate of one battery and it looked like 8 motors would overload it. Please let me know your thoughts on this.

**Reply by tobbeanton** [reply]
For testing purposes it should work fine. Worst thing the protection circuit kicks in and cuts the power.
